### ISSUE TITLE
"Open Recent" doesn't show last closed file.

### ISSUE THREAD

**Issue by calroc** [microsoft/vscode#88618]
- VSCode Version: Version: 1.41.1 (system setup)
Commit: 26076a4de974ead31f97692a0d32f90d735645c0
Date: 2019-12-18T14:58:56.166Z
Electron: 6.1.5
Chrome: 76.0.3809.146
Node.js: 12.4.0
V8: 7.6.303.31-electron.0
OS: Windows_NT x64 10.0.17134

Steps to Reproduce:

1. Create new file.  Add text. Save and close.
3. From menu select "File -> Open Recent >".
4. File from step 1 does not appear.  😞
5. From menu select "File -> Open File..." and open file from step 1.
6. From menu select "File -> Open Recent >" again, now the file appears.

Expected behaviour: file should appear in the "Open Recent" list after its editor is closed in step 1.

Does this issue occur when all extensions are disabled?: I didn't check.

**Comment by bpasero** [comment]
The list of File > Recent is only filed from files opened externally or from the File menu.

**Comment by calroc** [comment]
Hi,

The UI flow that's problematic is when you have a file that you have created and worked on, then closed, but never yet opened.  At that point, you won't see it in the "Open Recent" menu.

The way I first noticed this was after I had quit and restarted VSCode and wanted to continue working on a file, but it wasn't in the "Open Recent" menu, even though it was the last file I had worked on.

I now understand that that was because I hadn't _opened_ the file ever yet, just created, edited, and saved it.

My suggestion is that you add _new_ untitled files to the "Open Recent" menu when they are saved and titled.
